plugins/kubernetes/app/models/watchers/base_cluster_watcher.rb

```diff
@@ -8,8 +8,9 @@ class BaseClusterWatcher
 
     finalizer :stop_watching
 
-    def initialize(watch_stream)
-      @watch_stream = watch_stream
+    def initialize(args)
+      @cluster = args[:cluster]
+      @watch_stream = args[:watch_stream]
```

Comment:
I'm a bit confused here. The initializer expects a hash, while the `supervise` call below passes in just the cluster. How does that even work? :)